---
title: Helpful NullPointerExceptions
description: In this post we will learn how to use the new feature that show us more details on exceptions
categories:
 - tutorial
tags: JDK-14
---

Simple classes below.

```java
package dev.artsman.blog;

class Main {

  public static void main(String[] args) {
    var onewheel = new Onewheel();
    var brand = onewheel.tire().brand();
  }

}
```

```java
package dev.artsman.blog;

class Onewheel {
  private Tire tire;

  Tire tire() {
    return this.tire;
  }

}
```

```java
package dev.artsman.blog;

class Tire {
  private String brand;

  String brand() {
    return this.brand;
  }

}
```

Before JDK 14 the Null Pointer Exception it was like:


>Exception in thread "main" java.lang.NullPointerException
>	at dev.artsman.blog.Main.main(Main.java:7)


With JDK 14, the same exception looks like:


>Exception in thread "main" java.lang.NullPointerException: Cannot  invoke "dev.o1rb.study.jdk.Tire.brand()" because the return value of "dev.artsman.blog.Onewhell.tire()" is null at dev.artsman.blog.Main.main(Main.java:7)

It's because of the JEP 358: Helpful NullPointerExceptions(https://openjdk.java.net/jeps/358).<br />
This JEP 358 improve the usability of NullPointerExceptions generated by the JVM by describing precisely which variable was null.

Only need add this command line when run `-XX:+ShowCodeDetailsInExceptionMessages` and enjoy.<br />
*eof.*